### PR TITLE
Address remaining comments from query builder PR 

### PR DESCRIPTION
**Pull request by attfarhan**

### CODE REVIEW

File: shared/src/util/markdown.ts [hunk 1]
```diff
@@ -40,46 +40,22 @@ export const highlightCodeSafe = (code: string, language?: string): string => {
 /**
  * Renders the given markdown to HTML, highlighting code and sanitizing dangerous HTML.
  * Can throw an exception on parse errors.
+ *
+ * @param inlineCode whether to use inlineCode mode, which suppresses the default behavior of wrapping elements in <p> tags,
```
Comment (felixfbecker) [inline shared/src/util/markdown.ts]: Could you explain why query builder needs different behaviour? Especially omitting code highlighting makes little sense to me (it shouldn't highlight unless there is a language tag on it), the rest seems like things that can be easily controlled with CSS

File: web/src/search/input/QueryBuilder.tsx [hunk 2]
```diff
@@ -317,10 +317,13 @@ export class QueryBuilder extends React.Component<Props, QueryBuilderState> {
 
     private toggleShowQueryBuilder = (e: React.MouseEvent<HTMLAnchorElement>) => {
         e.preventDefault()
-        this.setState({ showQueryBuilder: !this.state.showQueryBuilder })
+        this.setState((prevState: QueryBuilderState, props: Props) => ({
```
Comment (felixfbecker) [inline web/src/search/input/QueryBuilder.tsx]:
```suggestion
        this.setState(prevState => ({ showQueryBuilder: !prevState.showQueryBuilder }))
```

File: web/src/search/input/QueryBuilderInputRow.tsx [hunk 3]
```diff
@@ -48,7 +53,8 @@ export const QueryBuilderInputRow: React.FunctionComponent<Props> = props => {
                     autoCapitalize="off"
                     autoComplete="off"
                     placeholder={placeholder}
-                    onChange={props.onInputChange(props.shortName)}
+                    // tslint:disable-next-line:jsx-no-lambda
+                    onChange={e => props.onInputChange[props.shortName](e, props.shortName)}
```
Comment (felixfbecker) [inline web/src/search/input/QueryBuilderInputRow.tsx]: This still has the same issue (as shown by disabling the lint rule). The idea was
```suggestion
                    onChange={props.onInputChange[props.shortName]}
```

File: web/src/search/input/QueryBuilder.tsx [hunk 4]
```diff
@@ -358,6 +363,22 @@ export class QueryBuilder extends React.Component<Props, QueryBuilderState> {
...
+        language: this.onInputChange,
+        patterns: this.onInputChange,
+        exactMatch: this.onInputChange,
+        case: this.onInputChange,
+        author: this.onInputChange,
+        after: this.onInputChange,
+        before: this.onInputChange,
+        message: this.onInputChange,
+        count: this.onInputChange,
+        timeout: this.onInputChange,
```
Comment (felixfbecker) [inline web/src/search/input/QueryBuilder.tsx]: There isn't really in point in having this record if all the properties are the same function - the idea was that each property would be a function only for this field. You could keep the old `onInputChange` signature, then define this as
```ts
private fieldsChanged = {
  type: this.onInputChange('type'),
  ...
}
```

File: web/src/search/input/QueryBuilder.tsx [hunk 5]
```diff
@@ -358,6 +363,22 @@ export class QueryBuilder extends React.Component<Props, QueryBuilderState> {
             return { fields: newFields, builderQuery: fieldsQueryParts.join(' ') }
         })
     }
+
+    private fieldsChanged = {
```
Comment (felixfbecker) [inline web/src/search/input/QueryBuilder.tsx]: Move properties to top of class

File: shared/src/util/markdown.ts (hunk 1 above)
Comment (attfarhan) [reply shared/src/util/markdown.ts]: It's not that easy to control with CSS. E.g. I can set all `<p>` tags to be `display:inline`, but then I get things like this: image

Anyways, there's no good reason for elements to be wrapped in `<p>`   tags in many cases (I've run into this before), so could be useful elsewhere.

Comment (felixfbecker) [reply shared/src/util/markdown.ts]: Well the markdown spec says that anything that any content separated by an empty line is a paragraph (`<p>`). So if a paragraph is not desired at all in that position why is it in the markdown?

Comment (attfarhan) [reply shared/src/util/markdown.ts]: Actually I don't need to render the examples as markdown. I just wrap it in a code tag always. I removed the `inlineCode` param.

**Comment by attfarhan** [comment]
@felixfbecker I resolved the comments remaining, PTAL

File: web/src/search/input/InfoDropdown.tsx [hunk 6]
```diff
@@ -59,7 +59,7 @@ export class InfoDropdown extends React.Component<Props, State> {
                                                 <span className="text-muted small">{ex.description}: </span>
                                                 <code
                                                     dangerouslySetInnerHTML={{
-                                                        __html: renderInlineMarkdown(ex.value),
+                                                        __html: ex.value,
```
Comment (felixfbecker) [inline web/src/search/input/InfoDropdown.tsx]: I'm confused, so we don't render the markdown at all anymore?

Comment (attfarhan) [reply web/src/search/input/InfoDropdown.tsx]: Yeah I realized we will _always_ want this to be an inline code snippet and nothing more/less. So `ex.value` just be a raw string, and we'll always render it as code.

Comment (felixfbecker) [reply web/src/search/input/InfoDropdown.tsx]: Where does the content for `ex.value` come from?

Comment (felixfbecker) [reply web/src/search/input/InfoDropdown.tsx]: So it's code, but not HTML, right? Because this still assigns it to `dangerouslySetInnerHTML`

Comment (attfarhan) [reply web/src/search/input/InfoDropdown.tsx]: Comes from the [examples prop](https://github.com/sourcegraph/sourcegraph/pull/2338/files/62a631d41f1f8fcd5b039aa373e531dc5129d0c3#diff-711fa2233cc9a815b12d2732b79c9eb3L14).

correct, just fixed that .

**Comment by felixfbecker** [comment]
Thanks for following up!

File: web/src/search/input/QueryBuilderInputRow.tsx [hunk 7]
```diff
@@ -48,7 +50,8 @@ export const QueryBuilderInputRow: React.FunctionComponent<Props> = props => {
                     autoCapitalize="off"
                     autoComplete="off"
                     placeholder={placeholder}
-                    onChange={props.onInputChange(props.shortName)}
+                    // tslint:disable-next-line:jsx-no-lambda
```
Comment (felixfbecker) [inline web/src/search/input/QueryBuilderInputRow.tsx]:
```suggestion
```

**Pull request merged**